mysql-test/suite/binlog/t/binlog_recover_async_rollback_trx.test

```diff
@@ -0,0 +1,82 @@
+--source include/have_debug.inc
+--source include/have_innodb.inc
+--source include/have_binlog_format_row.inc
+
+CREATE TABLE t1 (id int primary key, data int) ENGINE = InnoDB;
+INSERT INTO t1 VALUES (0, 1);
+
+--echo #
+--echo # 1. Check DML in prepared state can rollback correctly.
+--echo #
+
+--source include/expect_crash.inc
+SET debug = "d,crash_before_writing_xid";
+--error 2013
+UPDATE t1 SET data = data + 1 WHERE id = 0;
```

Comment:
It would be better to hold up the server with `DEBUG_SYNC` in a connection and then invoke `source include/kill_mysqld.inc` in `connection default`.

This test does not seem to work with `./mtr --ps-protocol`.